ringersong/src/main/java/com/RingerSong/free/data/SpotifyDownloaderRepository.kt

```
@@ -42,66 +44,88 @@ class SpotifyDownloaderRepository(private val context: Context) {
                 }
-            }
-
-            Log.d(TAG, "Downloaded: ${localFile.absolutePath}")
-            return@withContext localFile.absolutePath
 
-        } catch (e: Exception) {
-            Log.e(TAG, "Download error", e)
-            return@withContext null
+                Log.d(TAG, "Downloaded: ${localFile.absolutePath}")
+                DownloadResult.Success(localFile.absolutePath)
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)